### DISCUSSION TITLE
LAYOUTv2: Budgets and "Left to Spend" are not correctly calculated

### DISCUSSION THREAD

**Discussion by JanFeld** [firefly-iii/firefly-iii#4539] [General]
**Bug description**
I am running Firefly III version 5.5.1, and my problem is:

In Layout v2 the _Left to Spend_ as well as the view for the budgets are not correctly displayed.
When i'm using the old layout i've got ~250 € left, in the new layout it's about 700 €. The panel for the 'Custom timed Budgets' shows the current month as well as the last month. Maybe there's a relation between both. The bars indicating the spent and overspent amounts are shown for both months. To the right of the bars i can see how much i have left to spend this month.

Below the bar-section there are all budgets listed again wihtout any bars but show me the amount i've spent so far this month. I'm not sure if this is how it's supposed to be.

![grafik](https://user-images.githubusercontent.com/22739965/112749098-68d1ba00-8fc0-11eb-896f-ff35590b8622.png)

Additionally i'm not sure i'm understanding the Category view. In the picture below you can see that both amounts (outgoing and income) are the same. The bars don't really seem to make sense here. But that's another topic.

![grafik](https://user-images.githubusercontent.com/22739965/112749060-24461e80-8fc0-11eb-97fc-82607b33f5ae.png)

**Steps to reproduce**
Switch from layout v1 to v2. 

I know, it's pretty new. I just want to take notes! I couldn't try this on the demo site, the `?layout=v2` didn't work for me anymore.

Regards,
Jan

**Comment by JC5** [comment]
Alright I turned this into a discussion since it's all very much beta etc. Thanks for reporting the issue. I'm aware that both the budget and category overviews aren't really great yet.

What's the date range for the page? Is it what you expect it to be? Check it out under the dragon-menu on the right.

**Reply by JanFeld** [reply]
My timerange ist set to March 1st to March 31st, so i'd expect just seeing the budgets for march. 

**EDIT**: When set to the last seven days or a custom timerange the february-bars disappear! It's just the whole month 1st to 31st.

I really like the new layout! Would love to see it soon ;-)

**Reply by JC5** [reply]
Mmm. I don't see February in my list. But I know that 5.5.1 also brings some changes to how budgets are (automatically) generated. It may fix itself next month.

Glad to hear you like it! I'm working hard to get everything in place :+1:

**Comment by JC5** [comment]
I do see some issues with filtering (double budgets) which are fixed now. The category bars match my own monthly report and individual category overviews. The amounts are not the same (spent vs earned).

**Reply by JanFeld** [reply]
Oh, the category bars as well as the budget-left bars seem to change depending on the time range too. When selecting a shorter timerange (2nd to 31st) the budget left to spend as well as the bar for the categries change. It seems to be something with the selected timerange! Even though it's not displaying the correct amount. 

**EDIT:** Sorry, ignroe the last sentence. I've seen that behaviour in the v1 too. The _Left to spend_ changes to _Spent_ when selecting a custom timerange! Sorry for that. The amounts are the same and correct.

**Reply by JC5** [reply]
Glad to hear my code isn't entirely terrible 😉 . Let me know what you run into!
